### PR TITLE
addProfile function

### PR DESCRIPTION
**Pull request by dlebedynskyi**
just a function. 
https://github.com/noonereally/NW-Profession-Bot/issues/19
https://github.com/Phr33d0m/NW-Profession-Bot/pull/72

### CODE REVIEW

**Comment by dlebedynskyi** [comment]
ok this should be cleaner now. @BigRedBrent  @noonereally

**Comment by WloBeb** [comment]
Do you know, that all ""20->25 gather"" profiles will be to remove, and new profiles "mass gathering" should be added?
BTW. if you define profile level eg. 

<pre>23 : ["Crafted_Med_Armorsmithing_T4_Refine_Basic_Mass", "Crafted_Med_Armorsmithing_T4_Gather_Basic_Mass"],</pre> 

and you are out of ingredients for "Crafted_Med_Armorsmithing_T4_Refine_Basic_Mass" normal "Crafted_Med_Armorsmithing_T4_Gather_Basic" (without <b>_Mass</b>) will be started.

**Comment by WloBeb** [comment]
@dlebedynskyi I have only one little question: is the function addProfile() will work if the 'default' profile will be recursive?

And please stop  sending new "pull requests" until @Phr33d0m or @noonereally does not merge :D

**Comment by dlebedynskyi** [comment]
@WloBeb  do you know that I just wrote function and refactored  existing code without changes? 
anyway good to know. You can adjust profiles after. I think it is one of you guys who created gather profiles 
And please look at how many pull requests I send -it is really helpful before saying something really) As far as im aware there is only 2 active now. You do know how to look for it right? or how to use blame? 

edit. @WloBeb  yes it will.

**Comment by WloBeb** [comment]
@dlebedynskyi  I'm not sure if this new addProfile() is correct. When I saw this:

<pre>      for(var i = 0; i<= maxLevel; i++){
          //override
          if (profile.level && profile.level[i]){
              newProfile.level[i] = profile.level[i];
              continue;
          }
          //iterate and set
          if (newProfile.recursiveList && i>0){
              newProfile.level[i] = newProfile.level[i-1];
          }
      }</pre>

I have 2 questions:
1. What happend if 'base' profile is recursive and new 'profile' isn't? I think it could lead to profile with recursiveList set to false, and level[i] == undefined (for undefined levels of 'base' profile).
<s>2. Am I right, that these function copy tasklist from lvl[i-1] if lvl[i] == undefined? If yes, this could lead  <b>after expand</b> to very, very long tasklist.</s> EDIT: I change a way of expanding (no duplicate tasks).

In my opinion if 'base' profile is recursive <b>OR</b> new 'profile' is recursive this function should always produce recursive profile and copy everything from 'base' profile until new 'profile'.level[i] == undefined and copy 'profile' above.
<i><b>I admit that I have not tested this, but I read only code.</b></i>

**Comment by dlebedynskyi** [comment]
@WloBeb test it in jsFiddle really - https://jsfiddle.net/63f7mpf6/13/. 

1st - probably yes. In this case you do need to specify recursiveList for new profile. 
2nd. it will copy yes. it is limited to 25 levels by max Level constant.

I updated code for this cases. Setting recursive to true for new if is missing.

File: main.js [hunk 1]
```diff
@@ -662,30 +663,85 @@ function _select_Gateway() { // Check for Gateway used to
...
+      if (!definedTask[profession]){
+        definedTask[profession] = professionSet;
+      }  
+      
+      if (!profile) {return;}
+        
+      //getting base one to extend
+     var newProfile = jQuery.extend(true, {}, profileBase);
+     
+     if (!base) { base = (professionSet.taskListName === 'Leadership' ? 'XP' : 'default');}
```
Comment (Phr33d0m) [inline main.js:719]: Please avoid using 1-lined if() like that. Something like this instead is easier to read :-)

``` javascript
if (!base) 
    base = professionSet.taskListName === 'Leadership' ? 'XP' : 'default';
```

**Comment by Phr33d0m** [comment]
Lets test this out.

**Pull request merged**